storage/Devices/DeviceImpl.h

```diff
@@ -21,6 +21,32 @@ namespace storage
     class SystemInfo;
 
 
+    template <typename Type>
+    Type* to_device_of_type(Device* device)
+    {
+	static_assert(!is_const<Type>::value, "Type must not be const");
+
+	Type* tmp = dynamic_cast<Type*>(device);
+	if (!tmp)
+	    ST_THROW(DeviceHasWrongType("device has wrong type"));
```

Comment:
and thinking about it probably I  will switch argument so it reflect name ( first device, second arg Type)